scala-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/scala/ScalaInterpreter.scala

```diff
@@ -328,6 +466,10 @@ class ScalaInterpreter(private val config:Config = ConfigFactory.load) extends I
 
 object ScalaInterpreter {
 
+  val NamedResult = """(\w+):\s+([^=]+)\s+=\s*(.*)""".r
```

Comment:
Will this work when the result type contains an `=`, e.g. `res0: Int => Int = <function1>`?
Using a non-greedy match on anything preceding seems to work quite well `"""(?s)(res\d+):\s+(.+?)\s+=\s+(.*)""".r`